src/main/java/org/shanerx/tradeshop/TradeShop.java

```diff
@@ -49,364 +49,377 @@
+		toSave += addSetting("itrade-shop-name", "Server Shop");
+		toSave += addSetting("allow-double-trade", true);
+		toSave += addSetting("allow-quad-trade", true);
+		toSave += addSetting("max-edit-distance", 4);
+		toSave += addSetting("max-shop-users", 5);
+		toSave += addSetting("illegal-items", new String[]{"Bedrock", "Command_Block"});
+		toSave += addSetting("allow-custom-illegal-items", true);
+		toSave += addSetting("tradeshop-name", "Trade");
+		toSave += addSetting("itradeshop-name", "iTrade");
+		toSave += addSetting("bitradeshop-name", "BiTrade");
+		toSave += addSetting("allow-metrics", true);
+		toSave += addSetting("explode.trade", false);
+		toSave += addSetting("explode.itrade", false);
+		toSave += addSetting("explode.bitrade", false);
+
+		if (toSave > 0)
+			save();
+	}
+
+	public void addCustomItem(String name, ItemStack itm) {
+		if (!customItems.getValues(false).containsKey(name)) {
+			customItems.createSection(name);
+
+			customItems.set(name, itm.serialize());
+
+			save();
+		}
+	}
+
+	public void removeCustomItem(String name) {
+		if (customItems.getValues(false).containsKey(name)) {
+			customItems.set(name, null);
+
+			save();
+		}
+	}
+
+	public Set<String> getCustomItemSet() {
+		return customItems.getValues(false).keySet();
+	}
+
+	private void addCustomItemsDefaults() {
+
+		if (getCustomItems().getValues(false).isEmpty()) {
+			ItemStack dataHolder = new ItemStack(Material.TRIPWIRE_HOOK);
+			ItemMeta meta = dataHolder.getItemMeta();
+
+			meta.setDisplayName("Key");
+			meta.setLore(Arrays.asList("&aThe key to your dreams."));
+			dataHolder.setItemMeta(meta);
+
+			addCustomItem("Key", dataHolder);
+		}
+	}
+
+	public ItemStack getCustomItem(String name) {
+
+		ItemStack itm = null;
+		String cName = "";
+
+		for (String s : getCustomItems().getKeys(false)) {
+			if (s.equalsIgnoreCase(name)) {
+				cName = s;
+			}
+		}
+
+		if (cName.length() == 0) {
+			cName = name;
+		}
+
+		if (!(getCustomItems().get(cName) == null)) {
+			itm = ItemStack.deserialize(getCustomItems().getConfigurationSection(name).getValues(true));
+			ItemMeta meta = itm.getItemMeta();
+
+			if (meta.hasLore()) {
+				ArrayList<String> str2 = new ArrayList<>();
+				for (String s : meta.getLore()) {
+					str2.add(ChatColor.translateAlternateColorCodes('&', s));
+				}
+				meta.setLore(str2);
+			}
+
+			if (meta.hasDisplayName())
+				meta.setDisplayName(ChatColor.translateAlternateColorCodes('&', meta.getDisplayName()));
+
+			itm.setItemMeta(meta);
+		}
+
+		return itm;
+	}
+
+	public void save() {
+
+		if (messages != null)
+			try {
+				messages.save(messagesFile);
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+
+		if (settings != null)
+			try {
+				settings.save(settingsFile);
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+
+		if (customItems != null)
+			try {
+				customItems.save(customItemsFile);
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+
+		reloadConfig();
```

Comment:
Change Here

Reply:
Reloads the config which will re-run the add defaults however shouldn't save a second time unless settings were saved